Remove the #10088 hotfix for Teensy 3.1-like Input:Club keyboards

firetech opened the pull request:
## Description

On at least my Ergodox Infinity, the #10088 hotfix _causes_ the issue it intended to solve. That is, waking the keyboard from suspend after the keyboard has caused the host to wake works fine _without_ the #10088 hotfix, but _with_ it, the keyboard gets stuck sleeping.

This PR removes the #10088 hotfix for Ergodox Infinity, K-Type and Whitefox (all using the same MK20DX256 MCU).

## Types of Changes

- [x] Core
- [x] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

firetech commented:
I think _something_ should be done to fix this issue. Maybe not exactly this patch in its current state, but I'd be happy to revise it to something more acceptable.

drashna commented:
> I think _something_ should be done to fix this issue. Maybe not exactly this patch in its current state, but I'd be happy to revise it to something more acceptable.

How about making the function weak?  

That way, could throw a revised version in the boards.c in /platforms/chibios/ or wherever.

firetech commented:
I implemented @drashna's suggestion instead. As a side effect, the fix now only affects Input:Club boards (specifically K-Type, Whitefox and Ergodox Infinity).

Since I found there was an existing `IC_TEENSY_3_1` (where IC is for Input:Club) board in platforms/chibios, only used by K-Type and Whitefox, I switched over Ergodox Infinity to that and implemented the empty `restart_usb_driver` override there. No idea why Ergodox Infinity wasn't using that board, but it seems to work fine after switching to it. A comment in K-Type's rules.mk suggests that the only difference between `PJRC_TEENSY_3_1` and `IC_TEENSY_3_1` is a hack related to the watchdog, and given these boards common pedigree, that might be of use on the Ergodox Infinity as well?
